cmd/addon-operator/main.go

```diff
@@ -151,29 +143,25 @@ func runHAMode(ctx context.Context, operator *addon_operator.AddonOperator) {
 			OnStartedLeading: func(ctx context.Context) {
 				err := run(ctx, operator)
 				if err != nil {
-					log.Info(err)
-					os.Exit(1)
+					log.Fatal(err)
 				}
 			},
 			OnStoppedLeading: func() {
 				log.Info("Restarting because the leadership was handed over")
 				operator.Stop()
-				os.Exit(1)
+				os.Exit(0)
 			},
 		},
 		ReleaseOnCancel: true,
-	})
-	if err != nil {
-		log.Error(err)
+	}); err != nil {
+		log.Fatal(err)
 	}
 
 	go func() {
 		<-ctx.Done()
 		log.Info("Context canceled received")
-		err := syscall.Kill(1, syscall.SIGUSR2)
-		if err != nil {
-			log.Infof("Couldn't shutdown addon-operator: %s\n", err)
-			os.Exit(1)
+		if err := syscall.Kill(1, syscall.SIGUSR2); err != nil {
```

Comment:
We can use [notify context](https://pkg.go.dev/os/signal#NotifyContext) to catch syscalls
```go
ctx, stopNotify := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
defer stopNotify()
```